packages/local_auth/local_auth/android/src/test/java/io/flutter/plugins/localauth/LocalAuthTest.java

```diff
@@ -20,45 +24,95 @@
-  }
 
-  @Test
-  public void onDetachedFromActivity_ShouldReleaseActivity() {
-    final Activity mockActivity = mock(Activity.class);
-    final ActivityPluginBinding mockActivityBinding = mock(ActivityPluginBinding.class);
-    when(mockActivityBinding.getActivity()).thenReturn(mockActivity);
+  private LocalAuthPlugin plugin;
+  @Mock private Context mockContext;
+  @Mock private Activity mockActivity;
+  @Mock private Lifecycle mockLifecycle;
+  @Mock private FlutterEngine mockFlutterEngine;
+  @Mock private DartExecutor mockDartExecutor;
+  @Mock private ActivityPluginBinding mockActivityBinding;
+  @Mock private MethodChannel.Result mockResult;
+  @Mock private FlutterPluginBinding mockPluginBinding;
 
-    Context mockContext = mock(Context.class);
-    when(mockActivity.getBaseContext()).thenReturn(mockContext);
+  @Before
+  public void setUp() {
+    plugin = new LocalAuthPlugin();
+    mockContext = mock(Context.class);
+    mockActivity = mock(Activity.class);
+    mockLifecycle = mock(Lifecycle.class);
+    mockFlutterEngine = mock(FlutterEngine.class);
+    mockDartExecutor = mock(DartExecutor.class);
+    mockActivityBinding = mock(ActivityPluginBinding.class);
+    mockResult = mock(MethodChannel.Result.class);
+    mockPluginBinding = mock(FlutterPluginBinding.class);
 
     final HiddenLifecycleReference mockLifecycleReference = mock(HiddenLifecycleReference.class);
+    when(mockActivity.getApplicationContext()).thenReturn(mockContext);
     when(mockActivityBinding.getLifecycle()).thenReturn(mockLifecycleReference);
-
-    final Lifecycle mockLifecycle = mock(Lifecycle.class);
     when(mockLifecycleReference.getLifecycle()).thenReturn(mockLifecycle);
-
-    final FlutterPluginBinding mockPluginBinding = mock(FlutterPluginBinding.class);
-    final FlutterEngine mockFlutterEngine = mock(FlutterEngine.class);
+    when(mockActivityBinding.getActivity()).thenReturn(mockActivity);
+    when(mockActivity.getBaseContext()).thenReturn(mockContext);
     when(mockPluginBinding.getFlutterEngine()).thenReturn(mockFlutterEngine);
-
-    DartExecutor mockDartExecutor = mock(DartExecutor.class);
     when(mockFlutterEngine.getDartExecutor()).thenReturn(mockDartExecutor);
 
-    final LocalAuthPlugin plugin = new LocalAuthPlugin();
+    plugin.onAttachedToEngine(mockPluginBinding);
+    plugin.onAttachedToActivity(mockActivityBinding);
+  }
+
+  @Test
+  public void isDeviceSupportedReturnsFalse() {
+    plugin.onMethodCall(new MethodCall("isDeviceSupported", null), mockResult);
+    verify(mockResult).success(false);
+  }
+
+  @Test
+  public void onDetachedFromActivity_ShouldReleaseActivity() {
+
     plugin.onAttachedToEngine(mockPluginBinding);
     plugin.onAttachedToActivity(mockActivityBinding);
     assertNotNull(plugin.getActivity());
 
     plugin.onDetachedFromActivity();
     assertNull(plugin.getActivity());
   }
+
+  @Test
+  public void getAvailableBiometrics_ShouldReturnUndefinedBiometrics() {
```

Comment:
These names should describe the condition as well as the result, otherwise they are hard to follow; as written the tests sound like they are asserting three different behaviors for the same situation (which makes no sense). E.g., `getAvailableBiometrics_withHardwareButNothingEnrolled_ShouldReturnUndefined`